File: src/ts/2._linked_list.ts
// Estructura de datos
/** Linked List */

class Nodos {
    valor: number;
    siguiente: Nodos | null;

    constructor(valor: number) {
        this.valor = valor;
        this.siguiente = null;
    }
}

class ListaEnlace {
    cabeza: Nodos | null;

    constructor() {
        this.cabeza = null;
    }

    agregarAlInicio(valor: number): void {
        const nuevo = new Nodos(valor);

        nuevo.siguiente = this.cabeza;
        this.cabeza = nuevo;
    }
}

const linked = new ListaEnlace();

linked.agregarAlInicio(24);
linked.agregarAlInicio(18);
linked.agregarAlInicio(16);

console.log(linked.cabeza?.valor);

File: src/python/2._linked_list.py
"""
@author: luis
Estructura de datos
Linked List
"""

from typing import Optional

class Nodo:

    def __init__(self, valor: int):
        self.valor = valor
        self.siguiente: Optional[Nodo]

    def __repr__(self) -> str:
         return f"Nodo({self.valor})"


class ListaEnlace:
    cabeza: Optional[Nodo]

    def __init__(self):
        self.cabeza: Optional[Nodo] = None

    
    def agregarAlInicio(self, valor: int) -> None:
        """Agrega un nuevo nodo al inicio de la lista enlazada."""
        nuevo = Nodo(valor)

        nuevo.siguiente = self.cabeza
        self.cabeza = nuevo

# MAIN
if __name__ == "__main__":
        linked = ListaEnlace()
        linked.agregarAlInicio(24)
        linked.agregarAlInicio(18)
        linked.agregarAlInicio(16)

        print(linked.cabeza.valor if linked.cabeza else None)